File: typescript/easy/1455.isPrefixOfWord.ts
function isPrefixOfWord(sentence: string, searchWord: string): number {
  const splitWord = sentence.split(" ");

  for (let i = 0; i < splitWord.length; i++) {
    if (splitWord[i].startsWith(searchWord)) {
      return i + 1;
    }
  }

  return -1;
}

console.log(isPrefixOfWord("i love eating burger", "burg")); // 4
// Input: sentence = "i love eating burger", searchWord = "burg"
// Output: 4
// Explanation: "burg" is prefix of "burger" which is the 4th word in the sentence.

console.log(isPrefixOfWord("this problem is an easy problem", "pro")); // 2

console.log(isPrefixOfWord("i am tired", "you")); // -1

File: python/easy/1455.isPrefixOfWord.py
class Solution:
    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        split_word = sentence.split(" ")

        for i, word in enumerate(split_word):
            if word.startswith(searchWord):
                return i + 1

        return -1


if __name__ == "__main__":
    solution = Solution()
    print(solution.isPrefixOfWord("i love eating burger", "burg"))  # 4
    print(solution.isPrefixOfWord("i am tired", "you"))  # -1
